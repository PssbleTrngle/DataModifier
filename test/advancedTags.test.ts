import createTestAcceptor from './mock/TestAcceptor'
import setupLoader from './shared/loaderSetup'

const { loader } = setupLoader({ advancedTags: true })

describe('creation of tag definitions for the advanced tag-loader mod', () => {
   it('can generate files using remove entries', async () => {
      const acceptor = createTestAcceptor()

      loader.tags.blocks.add('#minecraft:fire_resistant_logs', '#minecraft:logs')
      loader.tags.blocks.remove('#minecraft:fire_resistant_logs', '#minecraft:logs_that_burn')

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('data/minecraft/tags/blocks/fire_resistant_logs.json')).toMatchSnapshot(
         'created #minecraft:fire_resistant_logs'
      )
   })

   it('fails when trying to remove using predicates', () => {
      const message = 'advanced tag loader only accepts tag entries in removal'

      expect(() => {
         loader.tags.blocks.remove('#minecraft:fire_resistant_logs', () => true)
      }).toThrow(message)

      expect(() => {
         loader.tags.blocks.remove('#minecraft:fire_resistant_logs', /regex/)
      }).toThrow(message)
   })
})
