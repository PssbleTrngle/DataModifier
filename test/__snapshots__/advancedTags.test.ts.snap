// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creation of tag definitions for the advanced tag-loader mod can generate files using remove entries: created #minecraft:fire_resistant_logs 1`] = `
{
  "remove": [
    "#minecraft:logs_that_burn",
  ],
  "replace": false,
  "values": [
    "#minecraft:logs",
  ],
}
`;
